.github/workflows/benchmark_trigger.yml

```
@@ -0,0 +1,113 @@
+# When a benchmark label is added, this cancels and then re-runs the CI
+# workflow. When the workflow is re-run, it will pick up the latest PR
+# description and labels. See
+# https://github.com/openxla/iree/issues/10042#issuecomment-1449250094 for
+# more background.
+
+on:
+  pull_request_target:
+    types:
+      - labeled
+
+env:
+  PR_NUMBER: ${{ github.event.number }}
+  HEAD_SHA: ${{ github.event.pull_request.head.sha }}
+  # Target workflow that runs the benchmarks.
+  WORKFLOW_NAME: CI
+  BENCHMARK_LABEL_PREFIX: 'benchmarks:'
+
+jobs:
+  # Precondition check is a separate step because we can't put the concurrency
+  # constraint on it; otherwise the irrelevant labeling events will cancel the
+  # events that meet the preconditions. Even with cancel-in-progress = false,
+  # the queued events will still be cancelled.
+  check:
+    runs-on: ubuntu-20.04
+    outputs:
+      should-rerun: ${{ steps.precondition.outputs.should-rerun }}
+    env:
+      GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+    steps:
+      - name: "Checking labels"
+        id: precondition
+        run: |
+          echo "should-rerun=$(jq --raw-output \
+            --arg label_prefix ${BENCHMARK_LABEL_PREFIX} \
+            '.label.name | startswith($label_prefix)' \
+            ${GITHUB_EVENT_PATH})" >> "${GITHUB_OUTPUT}"
+
+  trigger:
+    needs: check
+    if: needs.check.outputs.should-rerun == 'true'
+    runs-on: ubuntu-20.04
+    permissions:
+      actions: write
+    env:
+      GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+    concurrency:
+      # This cancels queued and in-progress trigger runs for the same PR.
+      group: ${{ github.workflow }}-${{ github.event.number }}
+      cancel-in-progress: true
+    steps:
+      - name: "Checking out repository"
+        # This checkouts from the main branch instead of the pull request. See
+        # https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#pull_request_target
+        uses: actions/checkout@ac593985615ec2ede58e132d2e21d2b1cbd6127c # v3.3.0
+      - name: "Finding the previous workflow run"
+        id: find-workflow
+        env:
+          RUN_JSON: workflow-run.json
+        run: |
+          gh api "/repos/${GITHUB_REPOSITORY}/actions/runs" --method GET -F head_sha="${HEAD_SHA}" \
+            | jq --arg name "${WORKFLOW_NAME}" '.workflow_runs | map(select(.name == $name))[0]' \
+            > "${RUN_JSON}"
+          echo "workflow-url=$(jq --raw-output '.url' ${RUN_JSON})" >> "${GITHUB_OUTPUT}"
+          echo "run-status=$(jq --raw-output '.status' ${RUN_JSON})" >> "${GITHUB_OUTPUT}"
+      - name: "Cancelling the previous workflow run"
+        # If the workflow isn't completed, we need to cancel it first; otherwise
+        # the API can't rerun it.
+        if: |
+          steps.find-workflow.outputs.workflow-url != 'null' &&
+          steps.find-workflow.outputs.run-status != 'completed'
```

Can we use `fromJSON` and treat these as actual values? Would probably need a proper type for workflow-url instead of null, but that seems easy (suggested edit above using https://stedolan.github.io/jq/manual/#Alternativeoperator://)

```suggestion
        if: |
          fromJSON(steps.find-workflow.outputs.workflow-url) != null &&
          steps.find-workflow.outputs.run-status != 'completed'
```

---

SG. I think it still makes sense to use null as "there is no workflow"